lambdas/service/app.py

```diff
@@ -341,7 +345,17 @@ def service_config(self) -> ServiceConfig:
 
     @property
     def facets(self) -> Sequence[str]:
-        return sorted(self.service_config.translation.keys())
+        return list(self.field_types_by_filterable_facets.keys())
+
+    @cached_property
+    def field_types_by_filterable_facets(self) -> Mapping[str, FieldType]:
+        field_types_by_filterable_facets = dict()
+        for facet, location in self.service_config.translation.items():
+            field_type = self.repository_controller.service.field_type(app.catalog,
```

Comment:
The fact that this reaches all the way through the controller into the service is a smell. Maybe this wants to be a controller method.

Reply:
@hannes-ucsc `this` is a little ambiguous, did you want to relocate the property I wanted to add `field_types_by_filterable_facets` or the method `self.repository_controller.service.field_type`?